python/lsst/ip/diffim/utils.py

```diff
@@ -1080,36 +1084,115 @@ def detectDipoleSources(self, doMerge=True, diffim=None, detectSigma=5.5, grow=3
+    psfSize : `float` | `tuple` [`float`]
         The FWHM of the PSF computed at its average position.
         Returns the widths along the Y and X axes,
         or the average of the two if `average` is set.
+
+    See Also
+    --------
+    evaluateMeanPsfFwhm
     """
-    pos = psf.getAveragePosition()
-    image = psf.computeKernelImage(pos).array
-    peak = psf.computePeak(pos)
+    if position is None:
+        position = psf.getAveragePosition()
+    image = psf.computeKernelImage(position).array
+    peak = psf.computePeak(position)
     peakLocs = np.unravel_index(np.argmax(image), image.shape)
+    width = _sliceWidth(image, peak/2., peakLocs, axis=0), _sliceWidth(image, peak/2., peakLocs, axis=1)
+    return np.nanmean(width) if average else width
+
+
+def evaluateMeanPsfFwhm(exposure, average=True, *, fwhmExposureBuffer=None, fwhmExposureGrid=None):
+    """Get the median PSF FWHM by evaluating it on a grid within an exposure.
+
+    Parameters
+    ----------
+    exposure : `~lsst.afw.image.Exposure`
+        The exposure for which the mean FWHM of the PSF is to be computed.
+        The exposure must contain a `psf` attribute.
+    average : `bool`, optional
+        Set to return the average width over Y and X axes.
+    fwhmExposureBuffer : `float`, optional
+        Fractional buffer margin to be left out of all sides of the image
+        during the construction of the grid to compute mean PSF FWHM in an
+        exposure.
+    fwhmExposureGrid : `int`, optional
+        Grid size to compute the mean FWHM in an exposure
 
-    def sliceWidth(image, threshold, peaks, axis):
-        vec = image.take(peaks[1 - axis], axis=axis)
-        low = np.interp(threshold, vec[:peaks[axis] + 1], np.arange(peaks[axis] + 1))
-        high = np.interp(threshold, vec[:peaks[axis] - 1:-1], np.arange(len(vec) - 1, peaks[axis] - 1, -1))
-        return high - low
-    width = (sliceWidth(image, peak/2., peakLocs, axis=0), sliceWidth(image, peak/2., peakLocs, axis=1))
-    return np.mean(width) if average else width
+    Returns
+    -------
+    medianFwhm : `float` | `tuple` [`float`]
+        The median PSF FWHM on the exposure
+
+    Notes
+    -----
+    If either of the ``fwhmExposureBuffer`` or ``fwhmExposureGrid`` parameters
+    is `None`, the PSF FWHM is computed at the average position of the PSF.
+    This behavior is same as that of `getPsfFwhm` but has a helpful error
+    message that indicates how to avoid the error from occurring.
+
+    See Also
+    --------
+    getPsfFwhm
+    """
+
+    psf = exposure.psf
+
+    if fwhmExposureBuffer is None or fwhmExposureGrid is None:
+        try:
+            fwhm = getPsfFwhm(psf, average=average)
+        except InvalidParameterError as err:
+            _LOG.error("Unable to compute PSF FWHM at average position. Set fwhmExposureBuffer and "
+                       "fwhmExposureGrid parameters to avoid this error.")
+            raise err
+        return fwhm
```

Comment:
Another behavior related question: Should this by default try to do `getPsfFwhm` and try to fall back to the grid evaluation only if `getPsfFwhm` fails at the average position>

Reply:
I would remove the option for `fwhmExposureBuffer` and `fwhmExposureGrid` to be `None`, and the wrapper around `getPsfFwhm`. The way the defaults are set, these parameters can't be `None`, anyway, and I think it is cleaner to explicitly call `getPsfFwhm` or `evaluateMeanPsfFwhm`, since the values will be different.